src/command/process_update_task.rs

```diff
@@ -180,6 +202,132 @@ impl UpdateProcessor {
+                let user_hour = vec[0].parse::<i8>().unwrap();
+                let minutes = vec[1].parse::<i8>().unwrap();
+
+                let schedule_weather_task = ScheduleWeatherTask::builder()
+                    .user_hour(user_hour)
+                    .minutes(minutes)
+                    .offset(offset)
+                    .username(self.username.clone())
+                    .chat_id(self.chat.id)
+                    // Safe unwrap checked in process_time func.
+                    .default_city_id(self.chat.default_city_id.unwrap())
+                    .build();
+
+                self.return_to_initial().await?;
+
+                let minutes_pretty = if minutes < 10 {
+                    format!("0{}", vec[1])
+                } else {
+                    vec[1].to_string()
+                };
+
+                let text = format!(
+                    "Weather info scheduled every day at {}:{} UTC {}",
+                    vec[0], minutes_pretty, offset
+                );
+
+                self.send_message(&text).await?;
+
+                Ok(Some(schedule_weather_task))
+            }
+
+            Err(_) => self.not_valid_offset_message().await,
+        }
+    }
+
+    async fn not_time_message(&self) -> Result<(), BotError> {
+        self.cancel(Some(
+            "That's not a well formatted time, it has to be formatted with this format `hour:minutes` being hour a number in range [0,23] 
+            and minutes a number in range [0,59]. The command was cancelled"
+            .to_string(),
+        ))
+        .await
+    }
+
+    fn parse_time(hour_or_minutes: &str, max_range: i32, min_range: i32) -> i32 {
+        match hour_or_minutes.parse::<i32>() {
+            Ok(number) => {
+                if !(min_range..=max_range).contains(&number) {
+                    -1
+                } else {
+                    number
+                }
+            }
+            Err(_) => -1,
+        }
+    }
+
+    async fn process_time(&self) -> Result<(), BotError> {
+        // check if user has default city.
+        if self.chat.default_city_id.is_none() {
+            return self
+                .cancel(Some(
+                    "To use /schedule command default city must be set first.".to_string(),
+                ))
+                .await;
+        }
+
+        let vec: Vec<&str> = self.text.trim().split(':').collect();
+
+        if vec.len() != 2 {
+            return self.not_time_message().await;
+        }
+
+        let hour = match Self::parse_time(vec[0], 23, 0) {
+            -1 => return self.not_time_message().await,
+            number => number,
+        };
+
+        let minutes = match Self::parse_time(vec[1], 59, 0) {
+            -1 => return self.not_time_message().await,
+            number => number,
+        };
+
+        self.repo
+            .modify_selected(
+                &self.chat.id,
+                self.chat.user_id,
+                format!("{}:{}", hour, minutes),
```

Comment:
maybe you can store time in a separate field.

and the time of this field can be `Time`

Reply:
We have to talk this in detail.

Reply:
Not necessary.